addon/bootstrap.js

```diff
@@ -132,29 +141,24 @@ function createLog(name, levelWord) {
   return L;
 }
 
-async function chooseVariation() {
-  let toSet, source;
-  const sample = studyUtils.sample;
-
-  if (studyConfig.variation) {
-    source = "startup-config";
-    toSet = studyConfig.variation;
-  } else {
-    source = "weightedVariation";
-    // this is the standard arm choosing method
-    const clientId = await studyUtils.getTelemetryId();
-    const hashFraction = await sample.hashFraction(studyConfig.studyName + clientId, 12);
-    toSet = sample.chooseWeighted(studyConfig.weightedVariations, hashFraction);
-  }
-  log.debug(`variation: ${toSet} source:${source}`);
-  return toSet;
-}
-
 function getHomepage(){
   // get the homepage of the user
-  var homepage = Services.prefs.getComplexValue(PREF_HOMEPAGE, Ci.nsIPrefLocalizedString).data;
+  var homepage;
+  try {
+    homepage = Services.prefs.getComplexValue(PREF_HOMEPAGE, Ci.nsIPrefLocalizedString).data;
+  } catch (e) {
+    console.error("Error obtaining the homepage: ", e);
+    return null;
+  }
   // transform the homepage into a nsIURI. Neccesary to get the base domain
-  var homepageURI = Services.netUtils.newURI(homepage);
+  var homepageURI;
+  // TODO: FIX PROBLEM WITH MALFORMED URIS (google.es/)
+  try {
+    homepageURI = Services.netUtils.newURI(homepage);
+  } catch (e) {
+    console.error("Error creating URI from homepage string: ", e);
+    return null;
+  }
 
   var eTLD;
   if (homepage.startsWith("about:")) {
```

Comment:
As discussed with Georg and during last meeting, let's specifically look for the `about:home` page. Here's what we should do:

1. If it's an `about:home` page, use this value for reporting.
2. If it's an about page other than the previous one, use the "about:pages" value for reporting.
3. If it's a valid URI, report the eTLD.
4. Otherwise end the study.